### ISSUE TITLE
custom_jvp does not work when first jvp rule is None

### ISSUE THREAD

**Issue by fehiepsi** [google/jax#3389]
Here is a repro code
```python
from jax import grad, custom_jvp

@custom_jvp
def f(x, y):
  return x ** 2 * y

f.defjvps(None, lambda x_dot, primal_out, x, y: 2 * x * y * x_dot)
print(grad(f, 1)(2., 3.))
```

**Comment by mattjj** [comment]
Thanks for finding this, and the clear repro. #3390 should squash it!

Always a pleasure to see you pop up on the issue tracker, @fehiepsi :)

**Comment by fehiepsi** [comment]
Thanks for a quick fix, @mattjj! (this issue happened in PR #3385.)
